documents/coredocument.go

```diff
@@ -76,6 +76,11 @@ type CoreDocument struct {
 	Document coredocumentpb.CoreDocument
 }
 
+type CollaboratorsAccess struct {
```

Comment:
Why do we need this?
Can you comment this pls

Reply:
we need this to differentiate the types of collaborators we will be adding to the rules, when we integrate the request header as part of the ClientAPI improvements epic.

Reply:
Then, shouldn't this be part of the client protobufs instead?

Reply:
The structures exist in the client protobufs as well, under the message RequestHeader. but we need this structure here in the coredoc to convert from api wire format on the create/update calls to a usable payload for coredoc methods.

Reply:
I think we should'nt create these but instead re-use them. Else there will be converters for just this. And these converters will get tricky once we add the more fiels to the write_acls

Reply:
I think we should not expose clientapi definitions into internal methods. This way we can evolve the clientapi independent of the core logic and viceversa.